src/datanode/src/instance/sql.rs

```diff
@@ -89,58 +88,116 @@ impl Instance {
                     .await
             }
             Statement::Alter(alter_table) => {
-                let req = self.sql_handler.alter_to_request(alter_table)?;
-                self.sql_handler.execute(SqlRequest::Alter(req)).await
+                let name = alter_table.table_name().clone();
+                let (catalog, schema, table) =
+                    table_idents_to_full_name(&name, session_ctx.clone())?;
+                let table_ref = TableReference::full(&catalog, &schema, &table);
+                let req = self.sql_handler.alter_to_request(alter_table, table_ref)?;
+                self.sql_handler
+                    .execute(SqlRequest::Alter(req), session_ctx)
+                    .await
             }
             Statement::DropTable(drop_table) => {
                 let req = self.sql_handler.drop_table_to_request(drop_table);
-                self.sql_handler.execute(SqlRequest::DropTable(req)).await
+                self.sql_handler
+                    .execute(SqlRequest::DropTable(req), session_ctx)
+                    .await
             }
             Statement::ShowDatabases(stmt) => {
                 self.sql_handler
-                    .execute(SqlRequest::ShowDatabases(stmt))
+                    .execute(SqlRequest::ShowDatabases(stmt), session_ctx)
                     .await
             }
             Statement::ShowTables(stmt) => {
-                self.sql_handler.execute(SqlRequest::ShowTables(stmt)).await
+                self.sql_handler
+                    .execute(SqlRequest::ShowTables(stmt), session_ctx)
+                    .await
             }
             Statement::Explain(stmt) => {
                 self.sql_handler
-                    .execute(SqlRequest::Explain(Box::new(stmt)))
+                    .execute(SqlRequest::Explain(Box::new(stmt)), session_ctx)
                     .await
             }
             Statement::DescribeTable(stmt) => {
                 self.sql_handler
-                    .execute(SqlRequest::DescribeTable(stmt))
+                    .execute(SqlRequest::DescribeTable(stmt), session_ctx)
                     .await
             }
             Statement::ShowCreateTable(_stmt) => {
                 unimplemented!("SHOW CREATE TABLE is unimplemented yet");
             }
+            Statement::Use(db) => {
+                ensure!(
+                    self.catalog_manager
+                        .schema(DEFAULT_CATALOG_NAME, &db)
+                        .context(error::CatalogSnafu)?
+                        .is_some(),
+                    error::SchemaNotFoundSnafu { name: &db }
+                );
+
+                session_ctx.set_current_schema(db);
+
+                Ok(Output::RecordBatches(RecordBatches::empty()))
+            }
         }
     }
 }
 
+/// Converts maybe fully-qualified table name (`<catalog>.<schema>.<table>`) to tuple.
+fn table_idents_to_full_name(
```

Comment:
I think we should add a `table_name` mod for these helpers. It would be used in other places.

Reply:
planned with some other refactoring, after issue #559 is resolved. added a todo here.